src/common/libflux/message.c

```diff
@@ -419,7 +419,7 @@ int flux_msg_get_flags (const flux_msg_t *msg, uint8_t *fl)
 
 int flux_msg_set_private (flux_msg_t *msg)
 {
-    uint8_t flags;
```

Comment:
Just a tiny bit more info in the commit message explaining why this is a problem now, as opposed to before this PR(?) would be helpful.

Reply:
this fix was "preemptive" as it really appeared after some refactoring for #3617.  But I'll add a note of how it is "sensitive" to changes.

Reply:
When I'm perusing C code, I usually take a local variable initialization as a subconscious hint to watch out for a code path in which its value is used before set, so when there isn't one, it tends to make my brain do extra work.  For that reason, sometimes I'll clean up a function by _removing_ unnecessary initialization when it's obvious it's not needed.  I'm not sure if that's reasonable, or if I should reprogram myself :-)

Reply:
i'm actually the same way and am not entirely sure why this started popping up recently.  It's this style of code that was hit:

```
int flags;
if (get_flags (ctx, &flags) < 0)
   goto error;
```

i don't know why changes somehow triggered the warning :-(   Perhaps some other code changes led to an extra pass in the compiler and that's all that happened.